files/check_keepalive.rb

```diff
@@ -0,0 +1,97 @@
+module Sensu
+module Extension
+
+# 1) update per-check timestamps in redis
+# 2) have eventmachine periodic timer to check if anything timed out and
+#    generate a new "critical" event for every
+#
+# extension has access to:
+#  - logger     @logger
+#  - redis      @settings[:redis]
+#  - transport  @settings[:transport]
+
+class CheckKeepalive < Handler
+  def name
+    'check_keepalive'
+  end
+
+  def description
+    'emit per-check keepalive events'
+  end
+
+  def post_init
+    @redis    = @settings.redis
+    @em_timer = EventMachine::PeriodicTimer.new(10) { process_keepalives }
+  end
+
+  def stop
+    @em_timer.cancel
+    yield
+  end
+
+  def run(event)
+    client_name = event[:client][:name]
+    check_name  = event[:check][:name]
+
+    rkey = "ext:check_keepalive:checks"
+    hkey = "#{client_name}:#{check_name}"
+    hval = event[:check][:keepalive]
```

Comment:
This currently doesn't happen. When puppet removes a check, it simply restarts sensu-client with new configuration. No additional info is sent to sensu-server.

At yelp we have the auto-resolve script which does a diff and auto resolves any checks that are removed. That might help here? But the /resolve endpoint will just re-play the same data with the same keepalive.